Clean-up `FullscreenManager` and add spec references

nt1m opened the pull request:
#### 9d5cda618299e1c0da62878c9efb94da55bad26a
<pre>
Clean-up `FullscreenManager` and add spec references
<a href="https://bugs.webkit.org/show_bug.cgi?id=288285">https://bugs.webkit.org/show_bug.cgi?id=288285</a>
<a href="https://rdar.apple.com/145351421">rdar://145351421</a>

Reviewed by Alex Christensen.

This PR does not contain any C++ code changes, and purely moves code around.

- Re-orders methods into sections that clearly correspond to spec algorithms:
   - Enter fullscreen flow is now grouped in one section
   - Exit fullscreen flow is now grouped in one section
   - Event dispatching methods are now grouped
- Adds spec references.
- Moves code less relevant to the understanding of the code to the header file.

This will make the code easier to read and cross-reference with the spec, especially after it is refactored for site isolation.

* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::isFullscreenEnabled const):
(WebCore::FullscreenManager::willEnterFullscreen):
(WebCore::FullscreenManager::didEnterFullscreen):
(WebCore::FullscreenManager::isSimpleFullscreenDocument const):
(WebCore::FullscreenManager::exitRemovedFullscreenElement):
(WebCore::FullscreenManager::cancelFullscreen):
(WebCore::FullscreenManager::queueFullscreenChangeEventForDocument):
(WebCore::FullscreenManager::emptyEventQueue):
(): Deleted.
* Source/WebCore/dom/FullscreenManager.h:

Canonical link: <a href="https://commits.webkit.org/290874@main">https://commits.webkit.org/290874@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/91590071581cd959eb0404c7529d3a7488889ed0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91373 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/10908 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/406 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96342 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42075 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/93423 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11285 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19237 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/70160 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/27686 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94374 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/8597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/82759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/50486 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/8363 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/344 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41231 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/78683 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/351 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98344 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18538 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/13593 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/79183 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18793 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/78597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/78387 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/22904 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/259 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/11686 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14446 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18537 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/23822 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18251 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21708 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20017 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/dom/FullscreenManager.h:
```diff
@@ -128,6 +128,22 @@ class FullscreenManager final : public CanMakeWeakPtr<FullscreenManager>, public
 #endif
 };
 
+class CompletionHandlerScope final {
```
This could be a nested private class.  Either way, I think I should generalize CompletionHandlerCallingScope and get rid of it.

nt1m committed:
Clean-up `FullscreenManager` and add spec references
https://bugs.webkit.org/show_bug.cgi?id=288285
rdar://145351421

Reviewed by Alex Christensen.

This PR does not contain any C++ code changes, and purely moves code around.

- Re-orders methods into sections that clearly correspond to spec algorithms:
   - Enter fullscreen flow is now grouped in one section
   - Exit fullscreen flow is now grouped in one section
   - Event dispatching methods are now grouped
- Adds spec references.
- Moves code less relevant to the understanding of the code to the header file.

This will make the code easier to read and cross-reference with the spec, especially after it is refactored for site isolation.

* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::isFullscreenEnabled const):
(WebCore::FullscreenManager::willEnterFullscreen):
(WebCore::FullscreenManager::didEnterFullscreen):
(WebCore::FullscreenManager::isSimpleFullscreenDocument const):
(WebCore::FullscreenManager::exitRemovedFullscreenElement):
(WebCore::FullscreenManager::cancelFullscreen):
(WebCore::FullscreenManager::queueFullscreenChangeEventForDocument):
(WebCore::FullscreenManager::emptyEventQueue):
(): Deleted.
* Source/WebCore/dom/FullscreenManager.h:

Canonical link: https://commits.webkit.org/290874@main
